scraping/scraping/spiders/_base.py

```diff
@@ -74,34 +78,40 @@ def __init__(
                 scraped for each starting page. Defaults to None.
         """
 
+        if not self.name:
+            logger.error("It's necessary to set the Spider's 'name' attribute.")
+
+        super().__init__(name=self.name, **kwargs)
+        self.table_name: str = getattr(self, "table_name", self.name)  # type: ignore
+
+        merchant, country = self.name.rsplit("_", 1)
+        self.merchant = merchant
+        self.country = CountryType(country)
+
         # set default value
         self.request_timeout = getattr(self, "request_timeout", 0.5)
-        self.table_name: str = getattr(self, "table_name", self.name)  # type: ignore
         self.StartRequest = SplashRequest  # default StartRequest is set to SplashRequest
-
-        super().__init__(name=self.name, **kwargs)
+        self.source = self.merchant  # TODO: change when source is no more equal to merchant
```

Comment:
This feels bad 😄 
Is there the possibility to move this to the child spiders?
Having this in the base class is probably not a good idea..

Reply:
I would prefer moving this to each child-spider explicitly, to be honest, to avoid possible weird future situations.

Reply:
Ok. Should I create a function which sets the `source` based on the spiders `name` or should I set it explicitly in each spider e.g. for amazon_DE: `source = 'amazon'`?

Reply:
Maybe add a similar block as `if not self.name:` here and to each spider the following:
```
name = source = TABLE_NAME_SCRAPING_AMAZON_DE
```

WDYT?

Reply:
Should `source` include the country information? Because this is part of `TABLE_NAME_SCRAPING_AMAZON_DE`. I think we defined `source = merchant`, right?

Reply:
Oh you'r absolutely right, that's wrong 🙈 

Let's use a block like this in the base spider: 
```
if not self.source:
  error_message = "It's necessary to set the Spider's 'source' attribute."
  logger.error(error_message)
  raise ValueError(error_message)
```
and remove `self.source = self.merchant`

And in each child spider, we do this `source, _ = self.name.rsplit("_", 1)` of course after setting its name.

This allows us to set the `source` attribute for new spiders as necessary. Does this make sense? 😅

Reply:
Ok yes, sounds good 👍